### ISSUE TITLE
Accessing inner iframe

### ISSUE THREAD

**Issue by TrevorDev** [spenibus/cors-everywhere-firefox-addon#10]
I am trying to inspect an iframe's inner content which is of a different origin. I am following this post http://stackoverflow.com/questions/17088609/disable-firefox-same-origin-policy which recommends trying your plugin but it doesn't seem to work. 

Is it a goal of this plugin to support this scenario? If not, do you believe this functionality is possible with Firefox's plugin model or would you recommend trying to build a custom version of Firefox?

Thanks.

**Comment by spenibus** [comment]
TL;DR
-----
- Not supported.
- Unsure about possible solution through addon architecture.
- Custom build likely needed.

The long stuff
--------------
As a preface: I am not an expert on the topic.

This addon creates fake server responses to trick Firefox into allowing content from a domain to be loaded while being requested from another domain, regardless of the (supposedly restrictive) content policies of the servers involved. This overridden behaviour is aimed at AJAX requests ([`XMLHttpRequest`][2]).

However this does not apply to iframes, as they involve [`X-Frame-Options`][1] rather than [`Access-Control-*`][3].

Your issue involves DOM manipulation, which is beyond the scope of this addon and almost certainly of [`X-Frame-Options`][1].

I am not intimate enough with the inner workings of Firefox's addon architecture to say whether or not this limitation can be lifted. I would imagine it is possible but I can not guarantee it. Note also that addon support is being transitioned to new architectures that appear much less permissive than the old one used by this addon.

I don't dabble in custom binaries but my uninformed opinion leads me to believe it would probably be the best way.

[1]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
[2]: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest
[3]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#The_HTTP_response_headers

**Comment by TrevorDev** [comment]
Thanks for the detailed response 🥇. 

I found a workaround this with chromium using --disable-web-security flag so I will likely stick to that browser for now and look at building ff if I have a need in the future.
